opslevel/resource_opslevel_check_repository_grep.go

```diff
@@ -0,0 +1,102 @@
+package opslevel
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	opslevel "github.com/opslevel/opslevel-go/v2022"
+)
+
+func resourceCheckRepositoryGrep() *schema.Resource {
+	return &schema.Resource{
+		Description: "Manages a repository file check",
+		Create:      wrap(resourceCheckRepositoryGrepCreate),
+		Read:        wrap(resourceCheckRepositoryGrepRead),
+		Update:      wrap(resourceCheckRepositoryGrepUpdate),
+		Delete:      wrap(resourceCheckDelete),
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: getCheckSchema(map[string]*schema.Schema{
+			"directory_search": {
+				Type:        schema.TypeBool,
+				Description: "Whether the check looks for the existence of a directory instead of a file. Defaults to false",
+				ForceNew:    false,
+				Optional:    true,
+			},
+			"filepaths": {
+				Type:        schema.TypeList,
+				MinItems:    1,
+				Description: "Restrict the search to certain file paths.",
+				ForceNew:    false,
+				Required:    true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"file_contents_predicate": getPredicateInputSchema(false),
```

Comment:
Still need to work through this if we want it to default to type: exists.  I'm assuming we'd want to make it a special case for this resource vs modifying getPredicateInputSchema which would affect other resources.

Reply:
the "false" here means its "not required" IIRC  - so then I would assume we could do a `d.GetOK` kind of thing and if that returns false (IE not set) then instantiate a copy in a local variable with the "default"

Reply:
Yep, that's what I was going to work on but I wanted to also update the description that it's going to set a default and I can't do that without updating it in the function right?

Reply:
I guess I could pass in a description?